src/box-container/box-container.tsx

```diff
@@ -113,7 +114,11 @@ export const BoxContainer = ({
     };
 
     return (
-        <Container {...otherProps}>
+        <Container
+            {...otherProps}
+            onClick={isHeaderContainerClickable && onHandleClick}
```

Comment:
should also add check for `collapsible` here to avoid unnecessary click listen and rerender